pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -266,8 +268,17 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
             "FST index is currently only supported on STRING type columns");
         Preconditions.checkState(dictEnabledColumn,
             "FST index is currently only supported on dictionary-encoded columns");
-        _fstIndexCreatorMap.put(columnName, new LuceneFSTIndexCreator(_indexDir, columnName,
-            (String[]) indexCreationInfo.getSortedUniqueElementsArray()));
+        TextIndexCreator textIndexCreator;
+        if (_config.getFSTIndexType() != null
+            && _config.getFSTIndexType() == FSTType.NATIVE) {
```

Comment:
(minor) The first check is redundant
```suggestion
        if (_config.getFSTIndexType() == FSTType.NATIVE) {
```